### DISCUSSION TITLE
What is the sample granularity of PCM when I set time interval to sample to 1s?

### DISCUSSION THREAD

**Discussion by oleotiger** [opcm/pcm#601] [Q&A]
The command I run : `pcm 1 -nc  -i=10 -csv=test.log`

I set the interval to sample performance counters to 1s with command as shown above. What confused me is that the output in `test.log` shows that  domain TIME(ticks) is 3000 ( Frequence of CPU is 3.0GHz).

Timestamps at the first column shows that samples are obeying 1s interval behavior.

But the domain TIME(Ticks) shows 3000, which means that the time it takes is 3000 clockticks ( = 1us).

Why timestamps can't be consistent when the TIME domain?

**Comment by opcm** [comment]
other users wanted the metric expressed in "time stamp counter" cycles (TSC). TSC is well established time measure. But I agree it also makes sense to add TIME(ms) metric. Is this what you want?

**Comment by oleotiger** [comment]
According to the wiki [PCM Column Names Decoder Ring](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-pcm-column-names-decoder-ring.html), `TIME(ticks)` here means `Number of invariant clockticks`.  

As I set time interval as 1s, there should be 3G ticks during per interval where 1ticks ~ 0.33ns. So the value I expect on the domain `TME(ticks)` is 3e9. But it shows 3e3 to me. This is what really confuses me.

**Comment by opcm** [comment]
ok, I see @twillhal the documentation should mention the miliions multiplier.

**Comment by oleotiger** [comment]
So the value of TIME(ticks) should be multiplied by 1 million? Or the domain should be changed to TIME(1e6 ticks)?

**Comment by twillhal** [comment]
My proposal would be to fix the documentation by mentioning the multiplier.

**Comment by oleotiger** [comment]
Should the value of `INST` be multiplied by 1million as well? In my experiment, the number of instructions is far blow my desired value.
